homework/index.js

```diff
@@ -15,32 +38,127 @@
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+          width: '50px',
+        });
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
     });
-    return elem;
   }
-
-  function main(url) {
+  // *************************************************
+  function ShowRepoInfo(arrayOfObjects, indexNumber) {
+    const wantedRepo = arrayOfObjects[indexNumber];
+    const infoContainer = document.getElementById('info_container');
+    infoContainer.innerHTML = '';
+    const infoLeft = createAndAppend('div', infoContainer, {
+      class: 'info-left',
+      text: 'Repository information:',
+    });
+    const nameAndLink = createAndAppend('p', infoLeft, {
+      class: 'p-info-left',
+      text: 'Name:',
+    });
+    createAndAppend('a', nameAndLink, {
+      class: 'repo-link',
+      text: wantedRepo.name,
+      href: wantedRepo.html_url,
+      target: '_blank',
+    });
+    const Description = createAndAppend('p', infoLeft, {
+      class: 'p-info-left',
+      text: 'Description:',
+    });
+    createAndAppend('span', Description, {
+      class: 'span-info-left',
+      text: wantedRepo.description,
+    });
+    const forks = createAndAppend('p', infoLeft, {
+      class: 'p-info-left',
+      text: 'Forks:',
+    });
+    createAndAppend('span', forks, {
+      class: 'span-info-left',
+      text: wantedRepo.forks,
+    });
+    const lastUpdate = createAndAppend('p', infoLeft, {
+      class: 'p-info-left',
+      html: 'Last&nbsp;update:',
+    });
+    createAndAppend('span', lastUpdate, {
+      class: 'span-info-left',
+      text: new Date(wantedRepo.updated_at).toLocaleString(),
+    });
+    ShowContributions(wantedRepo.contributors_url);
+  }
+  // *****************************************
+  function starter(url) {
     fetchJSON(url, (err, repositories) => {
       const root = document.getElementById('root');
+      const header = createAndAppend('header', root, {
+        class: 'header',
+      });
+      createAndAppend('span', header, {
+        class: 'page-title',
+        text: 'HYF Repositories',
+      });
+      const selectEl = createAndAppend('select', header, {
+        class: 'select-list',
+        id: 'repositories_list',
+      });
+      const infoContainer = createAndAppend('div', root, {
+        id: 'info_container',
+        class: 'info-container',
+      });
```

Comment:
If I'm not mistaken, you asked this question in class about where to put the code that is not dependent on the data you get from `fetchJSON()`. You could do it like this:

```js
function starter(url) {
  const root = document.getElementById('root');
  const header = createAndAppend('header', root, { class: 'header' });
  createAndAppend('span', header, {
    class: 'page-title',
    text: 'HYF Repositories',
  });
  const selectEl = createAndAppend('select', header, {
    class: 'select-list',
    id: 'repositories_list',
  });
  const infoContainer = createAndAppend('div', root, {
    id: 'info_container',
    class: 'info-container',
  });

  fetchJSON(url, (err, repositories) => {
    if (err) { 
   ...
```

Reply:
yes, exactly that was my question.  Thanks for the explanation. 
Best regards.